sru/modules.py

```diff
@@ -3,230 +3,78 @@
-                else:
-                    h_t = g_c_t * mask_c_ * reset_t
-                if mask_pad_ is not None:
-                    h_t = h_t * (1-mask_pad_[t])
-                h[t, :, di, :] = h_t
-
-            c_final.append(c_t.view(batch, d))
-        return h.view(length, batch, -1), torch.stack(c_final, dim=1).view(batch, -1)
-
+from sru.ops import (elementwise_recurrence_cpu,
+                     elementwise_recurrence_gpu,
+                     elementwise_recurrence_naive)
 
 class SRUCell(nn.Module):
     """
-    An SRU cell, i.e. a single recurrent neural network cell,
-    as per `LSTMCell`, `GRUCell` and `RNNCell` in PyTorch.
-
-    Args:
-        input_size (int) : the number of dimensions in a single
-            input sequence element. For example, if the input sequence
-            is a sequence of word embeddings, `input_size` is the
-            dimensionality of a single word embedding, e.g. 300.
-        hidden_size (int) : the dimensionality of the hidden state
-            of this cell.
-        dropout (float) : a number between 0.0 and 1.0. The amount of dropout
-            applied to `g(c_t)` internally in this cell.
-        rnn_dropout (float) : the amount of dropout applied to the input of
-            this cell.
-        use_tanh (bool) : use tanh activation
-        is_input_normalized (bool) : whether the input is normalized (e.g. batch norm / layer norm)
-        bidirectional (bool) : whether or not to employ a bidirectional cell.
+    A single SRU layer as per `LSTMCell`, `GRUCell` in Pytorch.
     """
 
+    __constants__ = ['input_size', 'hidden_size', 'output_size', 'rnn_dropout',
+                     'dropout', 'bidirectional', 'has_skip_term', 'highway_bias',
+                     'v1', 'rescale', 'activation_type', 'activation', 'custom_m',
+                     'projection_size', 'num_matrices', 'layer_norm', ]
+
     def __init__(self,
-                 input_size,
-                 hidden_size,
-                 dropout=0,
-                 rnn_dropout=0,
-                 bidirectional=False,
-                 n_proj=0,
-                 use_tanh=0,
-                 highway_bias=0,
-                 has_skip_term=True,
-                 layer_norm=False,
-                 rescale=True,
-                 v1=False,
-                 custom_m=None):
+                 input_size: int,
+                 hidden_size: int,
+                 dropout: float = 0.0,
+                 rnn_dropout: float = 0.0,
+                 bidirectional: bool = False,
+                 n_proj: int = 0,
+                 use_tanh: bool = False,
+                 highway_bias: float = 0.0,
+                 has_skip_term: bool = True,
+                 layer_norm: bool = False,
+                 rescale: bool = True,
+                 v1: bool = False,
+                 custom_m: Optional[nn.Module] = None):
+        """Initialize the SRUCell module.
```

Comment:
In general it's good to keep docstrings under 72 chars for parsing tools like sphynx, just FYI :)

Reply:
Gotcha. Will do.